File: practice/algorithms/implementation/sherlock_and_squares/SherlockAndSquares.java
import java.util.*;

public class SherlockAndSquares {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        for(int i = 0; i < tests; i++) {
            String[] parts = stdin.nextLine().split(" ");
            int lower = Integer.parseInt(parts[0]);
            int upper = Integer.parseInt(parts[1]);
            int squareNumbers;
            if(lower == upper) {
                if (Math.sqrt(lower) == Math.ceil(Math.sqrt(lower))) {
                    squareNumbers = 1;
                }
                else {
                    squareNumbers = 0;
                }
            }
            else {
                squareNumbers = (int)(Math.floor(Math.sqrt(upper)) - (Math.ceil(Math.sqrt(lower))) + 1);
            }
            System.out.println(squareNumbers);
        }
        stdin.close();
    }
}

File: practice/algorithms/implementation/sherlock_and_squares/sherlock_and_squares.rb
tests = gets.to_i
tests.times do
  (lower, upper) = gets.split(" ").map {|i| i.to_i}
  square_numbers = 0
  if(lower == upper)
    if (Math.sqrt(lower) == (Math.sqrt(lower)).ceil)
      puts 1
    else
      puts 0
    end
  else
    puts ((Math.sqrt(upper)).floor - (Math.sqrt(lower)).ceil + 1).to_i
  end
end
